src/zenml/integrations/openai/hooks/open_ai_failure_hook.py

```diff
@@ -102,11 +270,112 @@ def openai_alerter_failure_hook_helper(
                 f"Parameters: `{context.step_run.config.parameters}`",
                 f"Exception: `({type(exception)}) {exception}`",
                 "",
-                f"*OpenAI ChatGPT's suggestion (model = `{model_name}`) on how to fix it:*\n `{suggestion}`",
+                f"*OpenAI ChatGPT's suggestion (model = `{model_name}`) on how to fix it:*\n{suggestion}",
             ]
         )
 
-        alerter.post(message)
+        # If using SMTP Email alerter, use specialized formatting for better email display
+        if is_smtp_email_alerter:
+            # Create a plain text message for email body
+            plain_message = f"""OpenAI Failure Hook Notification - Step failed!
+
+Step: {context.step_run.name}
+Pipeline: {context.pipeline.name}
+Run: {context.pipeline_run.name}
+Stack: {client.active_stack.name}
+Parameters: {context.step_run.config.parameters}
+Exception: ({type(exception).__name__}) {exception}
+
+OpenAI ChatGPT's suggestion (model = {model_name}) on how to fix it:
+{suggestion}
+
+{plain_traceback}
+"""
+            # For HTML email, create a custom HTML body with proper formatting
+            html_body = f"""
+            <html>
```

Comment:
I did not verify in detail, but is there a way to remove some duplication in all these html strings in various places of this PR?

Reply:
Done